src/main/java/com/arpnetworking/metrics/jvm/collectors/PoolMemoryMetricsCollector.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2015 Groupon.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.jvm.collectors;
+
+import com.arpnetworking.metrics.Metrics;
+import com.arpnetworking.metrics.Units;
+import com.arpnetworking.metrics.jvm.ManagementFactory;
+
+import java.lang.management.MemoryPoolMXBean;
+import java.lang.management.MemoryType;
+import java.lang.management.MemoryUsage;
+import java.util.List;
+
+/**
+ * Collector class for JVM memory usage metrics for each memory pool. Uses the Java Management API to get the metrics
+ * data.
+ *
+ * @author Deepika Misra (deepika at groupon dot com)
+ * @author Brandon Arp (brandon dot arp at inscopemetrics dot com)
+*/
+// CHECKSTYLE.OFF: FinalClass - Allow clients to inherit from this.
+public class PoolMemoryMetricsCollector implements JvmMetricsCollector {
```

Comment:
Don't we also need to add registration for this to `JvmMetricsRunnable.Builder()`? It seems this is the one that should be used for both. No?

Reply:
Yep. Done.